### PR TITLE
expose wasm: updateAllocationWithRepair

### PR DESCRIPTION
**Pull request by boddumanohar**
### Changes
- exposes a function to updateAllocationWithRepair
- add a new gosdk function to get Filestats from a single blobber

### Fixes
- 

### Tests
Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/gosdk/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master AFTER merging this PR

### Associated PRs (Link as appropriate):
- blobber:
- 0chain:
- system_test: https://github.com/0chain/system_test/pull/633
- zboxcli: https://github.com/0chain/zboxcli/pull/401
- zwalletcli:
- Other: ...

### CODE REVIEW

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | add-replace-blobber  |
| gosdk  | update-alloc  |
| zboxcli  | add-replace-blobber  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Comment by service-0chain** [comment]
Manual system tests [cancelled] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | add-replace-blobber  |
| gosdk  | update-alloc  |
| zboxcli  | add-replace-blobber  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

File: zboxcore/sdk/downloadworker.go [hunk 1]
```diff
@@ -594,6 +594,41 @@ type blobberFile struct {
 	size           int64
 }
 
+func GetFileRefFromBlobber(allocationID, blobberId, remotePath string) (fRef *fileref.FileRef, err error) {
```
Comment (boddumanohar) [inline zboxcore/sdk/downloadworker.go:805]: I am using this function in system_tests.

File: zboxcore/sdk/downloadworker.go [hunk 2]
```diff
@@ -802,6 +803,41 @@ type blobberFile struct {
...
+	listReq.ctx = ctx
+	listReq.remotefilepath = remotePath
+
+	listReq.wg = &sync.WaitGroup{}
+	listReq.wg.Add(1)
+	rspCh := make(chan *fileMetaResponse, 1)
+	go listReq.getFileMetaInfoFromBlobber(listReq.blobbers[0], 0, rspCh)
+	listReq.wg.Wait()
+	resp := <-rspCh
+	return resp.fileref, nil
```
Comment (peterlimg) [inline zboxcore/sdk/downloadworker.go]: Let's return `resp.fileref, resp.err` as mentioned in another PR

**Review by peterlimg — CHANGES_REQUESTED** [review]

**Review by peterlimg — APPROVED** [review]
LGTM

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | false  |
| system-tests  | add-replace-blobber  |
| gosdk  | update-alloc  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0dns  | staging  |

**Pull request merged**